.github/workflows/python.yaml

```diff
@@ -72,4 +72,11 @@ jobs:
       with:
         version: ${{ env.CURRENT_VERSION }}
         alias: development
-        push: ${{ github.ref == 'refs/heads/main' }}  # build always, publish on 'main' only to prevent version clutter
+        push: ${{ github.ref == 'refs/heads/main' }} # build always, publish on 'main' only to prevent version clutter
+    - name: Execute jupytext in demo venv
+      run: |
+        python -m venv .demovenv
+        source .demovenv/bin/activate
+        pip install -r demos/requirements.txt
+        python demos/demo-ci/jupytext_to_executed_notebook.py ./demos/demo-ci/create_demo_notebook.py ./demos/demo_data_science_project.ipynb
+        rm -rf data .lakectl.yaml .demovenv
```

Comment:
Don't necessarily need to clean those up, I would rather put them under gitignore so the repo remains clean after executing this step.